src/sixty_four.rs

```diff
@@ -200,25 +206,34 @@ impl XxHash64 {
     }
 
     pub(crate) fn write(&mut self, bytes: &[u8]) {
-        let (unaligned_head, aligned, unaligned_tail) = bytes.as_u64_arrays();
-
-        if !self.buffer.is_empty() || !unaligned_head.is_empty() {
-            self.buffer_bytes(bytes);
-        } else {
-            self.core.ingest_chunks(aligned);
-            self.buffer_bytes(unaligned_tail);
+        let remaining = self.maybe_consume_bytes(bytes);
+        match remaining.len() {
+            0 => (),
+            // fastpath: no need to create UnalignedBuffer
```

Comment:
```suggestion
            // fast path: no need to create `UnalignedBuffer`
```

Reply:
Did you check the performance impact of not doing the special case?

Reply:
The current benchmarks do not expose this case. I'll add some and measure it.